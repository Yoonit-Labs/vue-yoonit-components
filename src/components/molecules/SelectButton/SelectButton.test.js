import { mount } from '@vue/test-utils'
import YooSelectButton from '@/components/molecules/SelectButton/SelectButton.vue'

const SlotText = 'Default Slot Text'

const mountComponent = () => {
  return mount(YooSelectButton, {
    slots: { default: SlotText },
    propsData: {
      rate: [
        {
          text: '1',
          value: 1
        },
        {
          text: '2',
          value: 2
        },
        {
          text: '3',
          value: 3
        }
      ]
    }
  })
}

describe('YooSelectButton Component', () => {
  let wrapper
  beforeEach(() => {
    wrapper = mountComponent()
  })

  it('Matches Snapshot', async () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Loads the Component HTML', async () => {
    expect(wrapper.classes('yoo-select')).toBe(true)
    expect(wrapper.find('button.yoo-btn').exists()).toBe(true)
  })

  describe('Props', () => {
    describe('multiple', () => {
      it('Has a valid default value', () => {
        expect(YooSelectButton.props.multiple.default).toBe(false)
      })
    })
  })

  describe('Events', () => {
    describe('doActive', () => {
      it('Emits doActive Event when multiple not active', async () => {
        wrapper.findAll('button.yoo-btn').at(2).trigger('click')
        wrapper.findAll('button.yoo-btn').at(0).trigger('click')
        expect(wrapper.emitted()).toHaveProperty('doClick')
        expect(wrapper.vm.selectedArray.length === 1)
      })

      it('Emits doActive Event when multiple active', async () => {
        await wrapper.setProps({ multiple: true })
        wrapper.findAll('button.yoo-btn').at(1).trigger('click')
        wrapper.findAll('button.yoo-btn').at(0).trigger('click')
        wrapper.findAll('button.yoo-btn').at(1).trigger('click')
        expect(wrapper.emitted()).toHaveProperty('doClick')
        expect(wrapper.vm.selectedArray.length === 1)
      })
    })
  }) // describe Events
})
