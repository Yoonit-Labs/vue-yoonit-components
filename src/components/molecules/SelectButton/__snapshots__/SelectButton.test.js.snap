// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooSelectButton Component Matches Snapshot 1`] = `
<div class="yoo-flex-layout yoo-select" style="flex-direction: row; flex-wrap: nowrap; justify-content: space-around; align-items: center; align-content: stretch;">
  <div class="yoo__container--button" height="inherit" style="height: inherit;"><button class="yoo-btn yoo-btn__fill--light yoo-btn__border--rounded yoo-btn--base yoo-btn--short yoo-btn--animation-none">
      <!---->
      <p class="yoo-btn__text yoo-btn__text--md yoo-btn__text--weight-semi-bold">1</p>
      <!---->
    </button></div>
  <div class="yoo__container--button" height="inherit" style="height: inherit;"><button class="yoo-btn yoo-btn__fill--primary yoo-btn__border--rounded yoo-btn--base yoo-btn--short yoo-btn--animation-none">
      <!---->
      <p class="yoo-btn__text yoo-btn__text--md yoo-btn__text--weight-semi-bold">2</p>
      <!---->
    </button></div>
  <div class="yoo__container--button" height="inherit" style="height: inherit;"><button class="yoo-btn yoo-btn__fill--light yoo-btn__border--rounded yoo-btn--base yoo-btn--short yoo-btn--animation-none">
      <!---->
      <p class="yoo-btn__text yoo-btn__text--md yoo-btn__text--weight-semi-bold">3</p>
      <!---->
    </button></div>
</div>
`;
